src/libraries/System.Security.AccessControl/src/System/Security/AccessControl/NativeObjectSecurity.cs

```diff
@@ -145,6 +146,14 @@ private static CommonSecurityDescriptor CreateInternal(ResourceType resourceType
                     {
                         exception = (name == null ? new FileNotFoundException() : new FileNotFoundException(name));
                     }
+                    else if (error == Interop.Errors.ERROR_PATH_NOT_FOUND)
+                    {
+                        exception = isContainer switch
+                        {
+                            false => (name == null ? new FileNotFoundException()      : new FileNotFoundException(name)),
```

Comment:
Can you not just pass a null message instead of this switch? The constructor parameter on the exception is nullable...

Reply:
Good point. I'll do that.